FirebaseMLModelDownloader/Sources/ModelDownloadTask.swift

```diff
@@ -116,7 +124,44 @@ extension ModelDownloadTask: URLSessionDownloadDelegate {
     }
 
     guard (200 ..< 299).contains(response.statusCode) else {
-      // TODO: Handle download url expiry + retries.
+      /// Possible failure due to download URL expiry.
+      if response.statusCode == 400 {
+        let currentDateTime = Date()
+        if currentDateTime > remoteModelInfo.urlExpiryTime {
+          /// Check if ok to retry download.
+          if numberOfRetries > 0 {
```

Comment:
Same here for guard, if that's allowed in order to reduce the nested statements.

Reply:
Yes it is! I made the change.